evalml/pipelines/components/transformers/samplers/oversamplers.py

```diff
@@ -0,0 +1,90 @@
+from evalml.pipelines.components.transformers.samplers.base_sampler import (
+    BaseOverSampler
+)
+
+
+class SMOTESampler(BaseOverSampler):
+    """SMOTE Oversampler component. Works on numerical datasets only. This component is only run during training and not during predict."""
+    name = "SMOTE Oversampler"
+    hyperparameter_ranges = {}
+
+    def __init__(self, sampling_ratio=0.25, sampling_ratio_dict=None, k_neighbors=5, n_jobs=-1, random_seed=0, **kwargs):
+        """Initialize the SMOTE Oversampler component.
+
+        Arguments:
+            sampling_ratio (float): This is the goal ratio of the minority to majority class, with range (0, 1]. A value of 0.25 means we want a 1:4 ratio
+                of the minority to majority class after oversampling. If the targets are multiclass, will create a dictionary using this ratio. Defaults to 0.25.
+            sampling_ratio_dict (dict): Dictionary which has keys corresponding to each class, and the values are the number of samples we want to oversample to for each class key.
```

Comment:
Wouldn't it be easier for users to specify the `sampling_ratio_dict` if they passed in the desired `sampling_ratio` rather than the desired counts?

My thought is that to use this properly in AutoMLSearch, users would need to know the size of the folds created by the data splitter, which isn't known ahead of time by most users and are also not equally sized. 

Even outside of search though, I think it's just easier to specify the desired "amount of sampling" when it's a percentage rather than an absolute count.

Reply:
mm I see, so you're thinking of passing 

```python
sampling_ratio_dict = {
    0: 1,
    1: 0.5,
    2: 0.4
}
```
versus 
```python
sampling_ratio_dict = {
    0: 1000,
    1: 500,
    2: 400
}
```
where `0` is the majority class?

My initial reasoning was that it'd be easier for users to know how many samples of each class exist rather than the percentage ratios between these classes and the majority class. Definitely see the value of using ratios, so maybe I can file a follow-up issue for this? @freddyaboulton

Reply:
@bchen1116 Sounds good! I think we should tackle that before adding the oversamplers to AutoMLSearch though. Just cause I don't think we can expect users to specify the counts correctly since they (most likely) don't know the split sizes ahead of time.

Reply:
That's a good observation, Freddy